File: RubyChallenges/order.rb
# https://www.codewars.com/kata/55c45be3b2079eccff00010f
# https://medium.com/@nothingisfunny/sorting-arrays-in-ruby-375c547a5d6

test = "is2 Thi1s T4est 3a"

def order(words)
    arr = words.split(' ')
    answer = arr.sort_by { |s| s.scan(/\d+/) }
    return answer.join(' ')
end


def ordering(words)
    arr = words.split(' ')
    answer = []
    count = 1

    while answer.length < arr.length
        arr.each do |word|
            if word.include?(count.to_s)
                answer << word
            end
        end
        count += 1
    end

    return answer.join(' ')
end

puts ordering(test)

File: JavaScript/order.js
// https://www.codewars.com/kata/55c45be3b2079eccff00010f

let input = ""; 

function order(words){
    let arr = words.split(' ');
    let answer = [];
    let count = 1;
    
    if (words == ""){
        return "";
    }
        
    else { 
        while (answer.length < arr.length){
            var pattern = new RegExp(count);
            for (let i = 0; i < arr.length; i++){
                if (pattern.test(arr[i])) {
                    answer[count - 1] = arr[i]
                }
            }
            count += 1;
        }
        return answer.join(' ')
    }
  
  }

  console.log(order(input));